src/AppRouter.js

```diff
@@ -7,8 +7,42 @@ import Profile from './components/profile/profile';
 import Quiz from './components/quiz/quiz';
 import logo from './img/logo.png'
 import './styles/nav.styles.css'
+import { auth, provider } from './services/firebase';
 
-function AppRouter() {
+class AppRouter extends Component {
+  constructor() {
+    super();
+    this.login = this.login.bind(this); 
+    this.logout = this.logout.bind(this);
+    this.state = {
+      user : null
```

Comment:
User has to be saved in session storage and all the methods which are related to checking whether user is logged in or not has to be moved to separate service.